github_connector_odoo/models/odoo_module_version.py

```diff
@@ -0,0 +1,507 @@
+            (type(info["author"]) == list)
+            and info["author"]
+            or (type(info["author"]) == tuple)
+            and [x for x in info["author"]]
+            or info["author"].split(",")
+        )
+        return {
+            "name": info["name"],
+            "technical_name": info["technical_name"],
+            "summary": info["summary"],
+            "website": info["website"],
+            "version": info["version"],
+            "license": info["license"],
+            "auto_install": info["auto_install"],
+            "icon": info["icon"],
+            "description_rst": info["description"],
+            "external_dependencies": info.get("external_dependencies", {}),
+            "author": ",".join([x.strip() for x in sorted(author_list) if x.strip()]),
+            "depends": ",".join([x for x in sorted(info["depends"]) if x]),
+            "repository_branch_id": repository_branch.id,
+            "module_id": module.id,
+            "category_id": self.get_module_category(info).id or None,
+        }
+
+    @api.model
+    def create_or_update_from_manifest(self, info, repository_branch, full_module_path):
+        module_obj = self.env["odoo.module"]
+        module_version = self.search(
+            [
+                ("technical_name", "=", str(info["technical_name"])),
+                ("repository_branch_id", "=", repository_branch.id),
+            ]
+        )
+
+        if not module_version:
+            # Create new module version
+            module = module_obj.create_if_not_exist(info["technical_name"])
+            vals = self.manifest_2_odoo(info, repository_branch, module)
+            vals["full_module_path"] = full_module_path
+            module_version = self.create(vals)
+
+        else:
+            # Update module Version
+            vals = self.manifest_2_odoo(
+                info, repository_branch, module_version.module_id
+            )
+            vals["full_module_path"] = full_module_path
+            module_version.write(vals)
+        icon_path = False
+        resize = False
+        if info.get("images"):
+            full_current_icon_path = os.path.join(
+                full_module_path, info.get("images")[0]
+            )
+            if os.path.exists(full_current_icon_path):
+                icon_path = full_current_icon_path
+        else:
+            for current_icon_path in self._ICON_PATH:
+                full_current_icon_path = os.path.join(
+                    full_module_path, current_icon_path, "icon.png"
+                )
+                if os.path.exists(full_current_icon_path):
+                    icon_path = full_current_icon_path
+                    resize = True
+                    break
+        if icon_path:
+            image_enc = False
+            try:
+                with open(icon_path, "rb") as f:
+                    image = f.read()
+                if resize:
+                    image_enc = tools.image_resize_image(
```

Comment:
I think that image_resize_image method does not exists...
see https://github.com/odoo/odoo/blob/559b3a464e42362002f3ed08d7dd25084fc2e175/odoo/tools/image.py#L265